### ISSUE TITLE
Image Resizer: Resolution always blurry

### ISSUE THREAD

**Issue by srikbe** [searene/Anki-Addons#11]
I don't know if I'm doing something wrong, but I've tried on a PC and a Mac and whenever I paste in an image it is always blurry with image resizer, even when the pic that i am pasting in was a higher resolution than other pix in Anki.  (I don't have this problem when I turn off the resizer.) I just want my images to be big enough to see on my iPhone but not huge. What am I doing wrong?  Sri.Kulkarni99@gmail.com
![screen shot 2017-11-02 at 5 12 06 pm](https://user-images.githubusercontent.com/24945299/32350721-89af471c-bff1-11e7-94f6-f14bf862d7c8.png)
![screen shot 2017-11-02 at 5 14 38 pm](https://user-images.githubusercontent.com/24945299/32350722-89b8d5e8-bff1-11e7-9cca-168d7f885e54.png)
![screen shot 2017-11-02 at 5 15 23 pm](https://user-images.githubusercontent.com/24945299/32350723-89d9a7b4-bff1-11e7-860e-5abe690692d5.png)

**Comment by searene** [comment]
Hi strikbe,

Could you try the following steps.

1. Configure ImageResizer the same as the following one.

![capture](https://user-images.githubusercontent.com/3351317/32359269-d2aa0f38-c087-11e7-8286-c61855f1662f.PNG)

2. Open this image in web browser(e.g. Chrome). This is a random image I found using Google Image.

https://pbs.twimg.com/profile_images/839721704163155970/LI_TRk1z_400x400.jpg

3. Right click on the image, select `Copy image`
4. Go back to Anki, right click on the editor, then paste this image.
5. How about the image quality now? For me it's good enough, could you paste your screenshot ?

![capture](https://user-images.githubusercontent.com/3351317/32359385-ae28b8e8-c088-11e7-8223-96228c00eaf8.PNG)

**Comment by srikbe** [comment]
Hi,

It's hard to see but yeah, it's more blurry.  I also attached some
screenshots of other images, of varying sizes, and they all come out
blurrier.  Does this not happen with your images?

**Comment by searene** [comment]
Hi,

No, for me, images' qualities are almost the same as original ones', only that they are smaller.
You can provide an image file/url, whose quality is obviously decreased after resizing. Then I will see if I can reproduce the problem, and try to find a way to solve it.

**Comment by srikbe** [comment]
![screen shot 2017-11-05 at 2 35 10 am](https://user-images.githubusercontent.com/24945299/32412858-4b090696-c1d2-11e7-8fbb-31341b1133f1.png)


 Hi,

Here's one.  http://t0.gstatic.com/images?q=tbn:ANd9GcT6uVh4D2K2ECAixCwUXhN9NM_x-RgChstnG_prDN_K5jCe36IiGyREVg

The image is clear when I do it without resizer, but with "resize on pasting" clicked it becomes very blurry.

**Comment by searene** [comment]
Hi,

Could you show your ImageResizer configurations?

**Comment by searene** [comment]
Hi,
Could you update ImageResizer and use the smooth transformation to see if it works?

**Comment by AbigaelElenara** [comment]
TL;DR: At some unspecified time after updating to Anki 2.1 and decreasing Anki's webscale, any images I resize - and previously resized images while I'm reviewing - are blurry. Normal images seem fine, except when I use Snipping Tool I notice a bit more blur.
I've tried all sorts of settings with Image Resizer, restarted Anki, edited the .bat file to 1.00... nothing.

I have been using Image Resizer successfully for a long time. Recently I updated to Anki 2.1 and had another issue where everything was too zoomed in, so I had to make a .bat file and make the web scale smaller. Yesterday I noticed these blurry images.
When I just paste images in, they look okay; I can't tell much difference in the resolution. But when I resize them, they turn blurry, which is worrying because I use Anki all day for homeschool and I can't leave the images full-size because of that over-zooming thing - they'd be even bigger than normal. Example without and with resizing:
<img width="775" alt="anki2" src="https://user-images.githubusercontent.com/33943723/55179268-16512f00-5155-11e9-9397-e7dd33b1ab61.png">
<img width="813" alt="anki" src="https://user-images.githubusercontent.com/33943723/55179275-18b38900-5155-11e9-9ed4-81632322e46c.png">
It's currently set to resize by height to 200 pixels, but I'd had it at 400 and 300 before with the same issues.
I just tried to edit the .bat file and reset the webscale back to 1.00 and see if that was the issue, but it's still doing the same thing!

This is happening even with files I resized a long time ago. I'm 99% sure this picture was crystal clear before (it looks veeery blurry in the full-sized window):
<img width="908" alt="file" src="https://user-images.githubusercontent.com/33943723/55179951-8f04bb00-5156-11e9-923b-b8e52a365bf2.png">

More experimentation: This MIGHT be a problem with images in general, just intensified by Image Resizer for some reason. I didn't resize this image, and it's definitely blurrier in Anki. That said, this is a screenshot, not a copy-pasted image - when I copy an image directly as I did above, I only notice the blurriness with IR.
<img width="427" alt="anki3" src="https://user-images.githubusercontent.com/33943723/55180701-1ef73480-5158-11e9-975e-018de6c54fe3.png">
<img width="427" alt="nada" src="https://user-images.githubusercontent.com/33943723/55180712-24547f00-5158-11e9-9cac-525aa0e513f6.png">
And the worst of all is when I do resize it:
<img width="383" alt="anki4" src="https://user-images.githubusercontent.com/33943723/55180754-39c9a900-5158-11e9-9b80-1d61d2344ce5.png">
